lib/tpm2.c

```diff
@@ -591,13 +592,36 @@ tool_rc tpm2_evictcontrol(ESYS_CONTEXT *esys_context,
     return tool_rc_success;
 }
 
+uint32_t fix_esys_hierarchy(TPMI_RH_HIERARCHY hierarchy)
+{
+#if defined(ESYS_3_0)
+    switch (hierarchy) {
+        case TPM2_RH_NULL:
+            return ESYS_TR_RH_NULL;
+        case TPM2_RH_OWNER:
+            return ESYS_TR_RH_OWNER;
+        case TPM2_RH_ENDORSEMENT:
+            return ESYS_TR_RH_ENDORSEMENT;
+        case ESYS_TR_RH_PLATFORM:
+            return ESYS_TR_RH_PLATFORM;
+        default:
+            return TSS2_ESYS_RC_BAD_VALUE;
+    }
+#elif defined(ESYS_2_3)
+    return hierarchy;
+#else
```

Comment:
@tstruk does this branch imply, versions lesser than 2.3? Does it mean also that versions prior to 2.3 did not support ESAPI?

Reply:
It is for tools >= 4.0, which require ESAPI >= 2.3.0.
This is inline with the configure check.